snap-common/etc/NetworkManager/NetworkManager.conf

```diff
@@ -10,4 +10,5 @@ dhcp=internal
 managed=false
 
 [device]
+keep-configuration=no
```

Comment:
Is this enough to solve the issue or do we also need the fix for the loopback device? I do not see the relantionship, although in any case I'm happy also with not taking control of the loopback, but I'm curious.

Reply:
Based on your knowledge sharing last daily I dug into NM's source code and figured out why it was not matching the connections. I have patched NM to allow matching based on my findings, and removed this config change